src/System.Data.Common/src/System/Data/DbSchemaFactory.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace System.Data.Common
+{
+    public abstract class DbSchemaFactory
+    {
+        abstract public List<DbColumn> GetColumnSchema(DbDataReader dataReader);
```

Comment:
Regardless of type (I agree with `ReadOnlyCollection<T>` here): `abstract public` => `public abstract` for consistency.